pkg/api/query/v1.go

```diff
@@ -1634,3 +1644,90 @@ func NewMetricMetadataHandler(client metadata.UnaryClient, enablePartialResponse
 		return t, warnings.AsErrors(), nil, func() {}
 	}
 }
+
+func (qapi *QueryAPI) tsdbStatus(r *http.Request) (any, []error, *api.ApiError, func()) {
+	span, ctx := tracing.StartSpan(r.Context(), "tsdb_statistics_query_request")
+	defer span.Finish()
+
+	ps := storepb.PartialResponseStrategy_ABORT
+	if qapi.enableStatusPartialResponse {
+		ps = storepb.PartialResponseStrategy_WARN
+	}
+
+	var (
+		warnings annotations.Annotations
+		err      error
+	)
+
+	limit, err := parseLimitParam(r.FormValue("limit"))
+	if err != nil {
+		return nil, nil, &api.ApiError{Typ: api.ErrorBadData, Err: err}, func() {}
+	}
+
+	if limit < 1 {
+		// Ensure that a positive limit is always applied.
+		limit = 10
+	}
+
+	var tenant string
+	if qapi.enforceTenancy {
+		tenant, err = tenancy.GetTenantFromHTTP(r, qapi.tenantHeader, qapi.defaultTenant, qapi.tenantCertField)
+		if err != nil {
+			return nil, nil, &api.ApiError{Typ: api.ErrorBadData, Err: err}, func() {}
+		}
+	}
+
+	if limit > math.MaxInt32 {
+		return nil, nil, &api.ApiError{Typ: api.ErrorBadData, Err: errors.Errorf("limit %d overflows int32", limit)}, func() {}
+	}
+	req := &statuspb.TSDBStatisticsRequest{
+		Tenant:                  tenant,
+		Limit:                   int32(limit),
+		PartialResponseStrategy: ps,
+	}
+
+	var stats map[string]*statuspb.TSDBStatisticsEntry
+	tracing.DoInSpan(ctx, "retrieve_tsdb_statistics", func(ctx context.Context) {
+		stats, warnings, err = qapi.status.TSDBStatistics(ctx, req)
+	})
+
+	if err != nil {
+		return nil, nil, &api.ApiError{Typ: api.ErrorInternal, Err: errors.Wrap(err, "retrieving tsdb statistics")}, func() {}
+	}
+
+	if tenant != "" {
+		return convertToTSDBSTatus(stats[tenant], limit), warnings.AsErrors(), nil, func() {}
+	}
+
+	// Merge statistics from all tenants.
+	aggregatedStats := &statuspb.TSDBStatisticsEntry{}
```

Comment:
So I thought about this a bit more. 

Right now, by default, we get the aggregate stats merged from all  TSDBs in multitsdb. And we can pass in a tenant to get specific tenant stats.

But there's also some utility in getting the map of tenants and tenant stats directly (powering a thanos tenant health dashboard)! 

Maybe it might be useful to have another bool URL query param here called `aggregate`?

Reply:
I agree that it would useful. But I'm not sure if it should be the same endpoint or if we need a dedicated one (given that the response data will look quite different).